### ISSUE TITLE
Support for MultiLabel MultiClass classification confusion matrix computation.

### ISSUE THREAD

**Issue by touqir14** [pytorch/ignite#1609]
## ❓ Questions/Help/Support

Does Ignite support multilabel multiclass classification confusion matrix computation? I was looking for something analogous to this tensorflow feature : https://www.tensorflow.org/addons/api_docs/python/tfa/metrics/MultiLabelConfusionMatrix .

**Comment by vfdev-5** [comment]
Thanks for asking! Currently, we do not provide such metric but it would make sense to add a feature request for this metric. Let's see if we could implement it...

**Comment by touqir14** [comment]
I once sent a PR to Pytorch's tnt project implementing this functionality : https://github.com/pytorch/tnt/pull/137 . Tnt's last stable release was back in 2019, so I am guessing they won't be releasing any new versions. Will a similar implementation work for ignite?

**Comment by vfdev-5** [comment]
Sounds great ! Yes, I think we could adapt this code to ignite's codebase keeping in mind the differences in the API and 
- computations using torch instead of numpy
- adaptation for distributed computation

Anyway, please send a draft PR and we can iterate over to finalize it. 
To get started, please check:
- [CONTRIBUTING](https://github.com/pytorch/ignite/blob/master/CONTRIBUTING.md)
- [How metrics work in Ignite](https://pytorch.org/ignite/metrics.html#how-to-create-a-custom-metric)

**Comment by touqir14** [comment]
Sure, I will send a draft PR within a day or two.
